boards/nordic/nrf52840dk/src/lib.rs

```diff
@@ -0,0 +1,930 @@
+//! | P0.05 | P6 3   | UART RTS |
+//! | P0.06 | P6 4   | UART TXD |
+//! | P0.07 | P6 5   | UART CTS |
+//! | P0.08 | P6 6   | UART RXT |
+//! | P0.11 | P24 1  | Button 1 |
+//! | P0.12 | P24 2  | Button 2 |
+//! | P0.13 | P24 3  | LED 1    |
+//! | P0.14 | P24 4  | LED 2    |
+//! | P0.15 | P24 5  | LED 3    |
+//! | P0.16 | P24 6  | LED 4    |
+//! | P0.18 | P24 8  | Reset    |
+//! | P0.19 | P24 9  | SPI CLK  |
+//! | P0.20 | P24 10 | SPI MOSI |
+//! | P0.21 | P24 11 | SPI MISO |
+//! | P0.22 | P24 12 | SPI CS   |
+//! | P0.24 | P24 14 | Button 3 |
+//! | P0.25 | P24 15 | Button 4 |
+//! | P0.26 | P24 16 | I2C SDA  |
+//! | P0.27 | P24 17 | I2C SCL  |
+
+#![no_std]
+#![deny(missing_docs)]
+
+use core::ptr::addr_of;
+
+use capsules_core::virtualizers::virtual_alarm::VirtualMuxAlarm;
+use capsules_extra::net::ieee802154::MacAddress;
+use capsules_extra::net::ipv6::ip_utils::IPAddr;
+use kernel::component::Component;
+use kernel::hil::led::LedLow;
+use kernel::hil::time::Counter;
+#[allow(unused_imports)]
+use kernel::hil::usb::Client;
+use kernel::platform::{KernelResources, SyscallDriverLookup};
+use kernel::scheduler::round_robin::RoundRobinSched;
+#[allow(unused_imports)]
+use kernel::{capabilities, create_capability, debug, debug_gpio, debug_verbose, static_init};
+use nrf52840::chip::Nrf52DefaultPeripherals;
+use nrf52840::gpio::Pin;
+use nrf52840::interrupt_service::Nrf52840DefaultPeripherals;
+use nrf52_components::{UartChannel, UartPins};
+
+// The nRF52840DK LEDs (see back of board)
+const LED1_PIN: Pin = Pin::P0_13;
+const LED2_PIN: Pin = Pin::P0_14;
+const LED3_PIN: Pin = Pin::P0_15;
+const LED4_PIN: Pin = Pin::P0_16;
+
+// The nRF52840DK buttons (see back of board)
+const BUTTON1_PIN: Pin = Pin::P0_11;
+const BUTTON2_PIN: Pin = Pin::P0_12;
+const BUTTON3_PIN: Pin = Pin::P0_24;
+const BUTTON4_PIN: Pin = Pin::P0_25;
+const BUTTON_RST_PIN: Pin = Pin::P0_18;
+
+const UART_RTS: Option<Pin> = Some(Pin::P0_05);
+const UART_TXD: Pin = Pin::P0_06;
+const UART_CTS: Option<Pin> = Some(Pin::P0_07);
+const UART_RXD: Pin = Pin::P0_08;
+
+const SPI_MOSI: Pin = Pin::P0_20;
+const SPI_MISO: Pin = Pin::P0_21;
+const SPI_CLK: Pin = Pin::P0_19;
+const SPI_CS: Pin = Pin::P0_22;
+
+const SPI_MX25R6435F_CHIP_SELECT: Pin = Pin::P0_17;
+const SPI_MX25R6435F_WRITE_PROTECT_PIN: Pin = Pin::P0_22;
+const SPI_MX25R6435F_HOLD_PIN: Pin = Pin::P0_23;
+
+/// I2C pins
+const I2C_SDA_PIN: Pin = Pin::P0_26;
+const I2C_SCL_PIN: Pin = Pin::P0_27;
+
+// Constants related to the configuration of the 15.4 network stack
+const PAN_ID: u16 = 0xABCD;
+const DST_MAC_ADDR: capsules_extra::net::ieee802154::MacAddress =
+    capsules_extra::net::ieee802154::MacAddress::Short(49138);
+const DEFAULT_CTX_PREFIX_LEN: u8 = 8; //Length of context for 6LoWPAN compression
+const DEFAULT_CTX_PREFIX: [u8; 16] = [0x0_u8; 16]; //Context for 6LoWPAN Compression
+
+/// Debug Writer
+#[allow(unused)] // TODO: this throws unused warnings in `make ci-job-capsules`
+pub mod io;
```

Comment:
Not necessarily going to block this PR for this — but is this something that could be fixed?

The debug infrastructure is a bit fragile with the off-common-path statics, and I'm a bit worried this is hiding a real issue.

Reply:
Fixed. I had a hard time tracking down the root cause of this issue; it turned out to be conditional compilation, of course. `make ci-job-capsules` apparently passes `cfg(test)`, which does not select our custom panic handler. The reason why this didn't trip CI before is that `ci-job-capsules` reall]y tests _every_ crate (beyond just capsules), but only _library_ ones. With this crate being promoted to a library crate, it was now being "tested" and hence raised unused warnings.